sentinel/sentinel.go

```diff
@@ -138,7 +152,7 @@ func NewClient(
 			return nil, &ClientError{err: err, SentinelErr: true}
 		}
 		addr := l[3] + ":" + l[5]
-		pool, err := pool.New("tcp", addr, poolSize)
+		pool, err := pool.NewCustom("tcp", addr, poolSize, (pool.DialFunc)(df))
```

Comment:
is it actually necessary to cast here? I feel like you could get away with:

```go
pool, err := pool.NewCustom("tcp", addr, poolSize, df)
```

like you do on line 241